script/gn-asar.js

```diff
@@ -46,16 +46,20 @@ try {
   }
 } catch (err) {
   console.error('Unexpected error while generating ASAR', err)
-  fs.removeSync(tmpPath)
-  process.exit(1)
+  fs.remove(tmpPath)
+    .then(() => process.exit(1))
+    .catch(() => process.exit(1))
+  return
 }
 
 // Create the ASAR archive
 asar.createPackageWithOptions(tmpPath, out[0], {}, (err) => {
-  fs.removeSync(tmpPath)
-
-  if (err) {
-    console.error('Unexpected error while generating ASAR', err)
-    process.exit(1)
+  const checkErr = () => {
+    if (err) {
+      console.error('Unexpected error while generating ASAR', err)
+      process.exit(1)
+    }
   }
+
+  fs.remove(tmpPath).then(checkErr).catch(checkErr)
```

Comment:
@MarshallOfSound it's dying here bc the version of fs-extra we get for free in `package-lock` is too old: only `fs-extra` > v3 has inbuilt support for promises. It's probably too much effort to elevate `fs-extra` just to get promise support so maybe just use a callback here?